### ISSUE TITLE
ordering in includes is broken.

### ISSUE THREAD

**Issue by steverweber** [saltstack/salt#47954]
pillar:

x.sls
```
include:
    - x3
    - x2
    - x1
```
x3.sls
```
x: 3
```
x2.sls
```
x: 2
```
x1.sls
```
x: 1
```

```
salt-run --log-level=debug pillar.show_pillar system_with_state.test.com
```

```
[DEBUG   ] Rendered data from file: /var/cache/salt/master/git_pillar/...
include:
    - x3
    - x2
    - x1
[DEBUG   ] Rendered data from file: /var/cache/salt/master/git_pillar/...//x1.sls:
[DEBUG   ] Rendered data from file: /var/cache/salt/master/git_pillar/...//x2.sls:
[DEBUG   ] Rendered data from file: /var/cache/salt/master/git_pillar/...//x3.sls:

x: 3
```
where x should be `1`!

the bug seems to be:
https://github.com/saltstack/salt/blob/develop/salt/pillar/__init__.py#L764

it is changing the matched_pstates array to a set:
```
for sub_sls in set(matched_pstates):
```
perhaps it was changed to this to remove duplicates...

fixes the issue.
```
for sub_sls in matched_pstates:
```

**Comment by steverweber** [comment]
this is kinda critical to get fixed and back-ported.

**Comment by Ch3LL** [comment]
i'm not able to replicate this on develop which is the version of salt you pointed to in the code. can you include a versions-report so i can attempte to replicate on the same version you are using.

Here is my return from salt-run

```
➜  salt git:(7de5dbc755) sudo salt-run pillar.show_pillar the_cake_is_a_lie
x:
    1
```

Do you have anything set in master config to change the order of pillar rendering?

**Comment by steverweber** [comment]
i'll need to do some more investigation on this..
noticed today that the fix didn't work ....
`salt-run pillar.show_pillar the_cake_is_a_lie`
is giving different ordering than
`salt the_cake_is_a_lie pillar.items`
really odd.

**Comment by Ch3LL** [comment]
ping @terminalmage might have some ideas here as well to help with the fix

**Comment by steverweber** [comment]
lets also add a test case along with the fix so we don't run into this one again.
I would create the test case myself a few months ago but had trouble figuring out how to write it.

**Comment by steverweber** [comment]
@Ch3LL I'll create a pull request because this is a problem I don't want to patch on my end for each release.

**Comment by cachedout** [comment]
We will want @RichardW42 to look at this since he made the original change. This seems like it should be safe but I would like to hear what he thinks.

**Comment by terminalmage** [comment]
Marking as closed since #48524 was merged.
